### ISSUE TITLE
tailscale-operator: `ExternalName` `Service` to an advertised route doesn't work

### ISSUE THREAD

**Issue by serhanekicii** [tailscale/tailscale#12322]
### What is the issue?

I can reach to `admin101` host as follows using an `ExternalName` `Service`. This host is a subnet router advertises `10.0.1.0/24` and `10.0.0.0/24`.

```
apiVersion: v1
kind: Service
metadata:
  annotations:
    tailscale.com/tailnet-fqdn: admin101.tailXXXXXX.ts.net
  name: admin101 # service name
spec:
  externalName: placeholder  # any value - will be overwritten by operator
  type: ExternalName
```

I know that using IP ranges are not supported in annotations yet as stated in [docs](https://tailscale.com/kb/1236/kubernetes-operator#limitations). However, I was expecting to reach subnet router advertised routes by creating an `ExternalName` `Service` pointing to an advertised route as follows but it seems like it doesn't work.

```
apiVersion: v1
kind: Service
metadata:
  annotations:
    tailscale.com/tailnet-ip: 10.0.1.6 
  name: ns101 # service name
spec:
  externalName: placeholder   # any value - will be overwritten by operator
  type: ExternalName
```

```
/ # ping admin101.tailscale.svc.cluster.local -c 1 # This is the subnet router.
PING admin101.tailscale.svc.cluster.local (10.244.1.170): 56 data bytes
64 bytes from 10.244.1.170: seq=0 ttl=62 time=13.042 ms

--- admin101.tailscale.svc.cluster.local ping statistics ---
1 packets transmitted, 1 packets received, 0% packet loss
round-trip min/avg/max = 13.042/13.042/13.042 ms
/ # ping ns101.tailscale.svc.cluster.local -c1 -t10 # This is the ExternalName Service pointing to advertised subnet.
PING ns101.tailscale.svc.cluster.local (10.244.1.226): 56 data bytes

--- ns101.tailscale.svc.cluster.local ping statistics ---
1 packets transmitted, 0 packets received, 100% packet loss
/ #
```

If this is a known limitation, it would be nice to see it in the docs.

### Steps to reproduce

Above section is descriptive enough to reproduce.

### Are there any recent changes that introduced the issue?

No.

### OS

Linux

### OS version

Linux 6.1.0-17-amd64

### Tailscale version

1.66.4

**Comment by irbekrm** [comment]
Hi @serhanekicii ,

by default the Linux client does not accept advertized routes from tailnet peers. The proxies that the operator creates are essentially a wrapping around the Linux client, so the behaviour is the same.
We do not want to accept routes by default (this could potentially cause issues for Kubernetes users with subnet routers that expose CIDR ranges that overlap with cluster Pod/Service CIDRs, for example).

We did , however recently add a way to optionally configure the proxies to accept routes via `ProxyClass`, see https://github.com/tailscale/tailscale/pull/12388.
This functionality is currently available on unstable track (Helm chart version "1.67.110", container version "unstable-v1.67.110") and will be released in stable in 1.68.

Do let us know if that solves your issue

**Comment by serhanekicii** [comment]
I haven’t had the chance to test it personally, but it seems like the referenced commits are addressing the issue.
